### PR TITLE
fix differenceInDays across DST (#533)

### PR DESCRIPTION
**Pull request by dcousens**
Resolves https://github.com/date-fns/date-fns/issues/533

I'm happy to make changes where necessary,  this was simply the simplest solution for the primitives.

### CODE REVIEW

**Comment by dcousens** [comment]
For reference,  before this patch,  the `differenceInDays` function returned the following in `TZ=Australia/Sydney`
``` js
console.log(differenceInDays(c, b))
console.log(differenceInDays(b, a)) // 23 hours < 1 day
console.log(differenceInDays(c, a)) // 47 hours < 2 days
```

And for `TZ=UTC`,  it returned the expected result.

**Comment by dcousens** [comment]
@kossnocorp anything I need to to help this?

File: src/differenceInDays/index.js [hunk 1]
```diff
@@ -42,15 +40,9 @@ export default function differenceInDays(dirtyDateLeft, dirtyDateRight) {
   var dateLeft = toDate(dirtyDateLeft)
   var dateRight = toDate(dirtyDateRight)
 
-  var sign = compareAsc(dateLeft, dateRight)
-  var difference = Math.abs(differenceInCalendarDays(dateLeft, dateRight))
+  var result = (dateLeft - dateRight) / 86400000
```
Comment (dkozickis) [inline src/differenceInDays/index.js]: Please move `86400000` into a `var MILLISECONDS_IN_DAY`

Comment (dcousens) [reply src/differenceInDays/index.js]: @dkozickis moved

**Comment by dcousens** [comment]
> Would be cool if you can add a bit info about the case fixed in this PR to the function description?

What about
> This function accounts for the number of 24 hour periods between two timestamps,  and thereby ignores DST changes.

**Comment by dkozickis** [comment]
@dcousens sound logical to me

**Pull request merged**